doc/cascadia/Cascading-Default-Settings.md

```diff
@@ -0,0 +1,399 @@
+---
+author: Mike Griese @zadjii-msft
+created on: 2019-05-31
+last updated: 2019-06-13
+issue id: #754
+---
+
+# Cascading Default + User Settings
+
+## Abstract
+
+This spec outlines adding support for a cascading settings model. In this model,
+there are two settings files, instead of one.
+
+1. The default settings file
+2. The user's settings file
+
+The default settings file would be a static, read-only file shipped with the
+terminal. The user settings file would then contain all the user's chosen
+customizations to the settings. These two files would then be composed together
+when the app is launched, so that the runtime settings are the union of both the
+defaults and whatever modifications the user has chosen. This will enable the
+app to always use a default schema that it knows will be valid, and minimize the
+settings that the user needs to customize.
+
+Should the settings schema ever change, the defaults file will change, without
+needing to re-write the user's settings file.
+
+
+## Inspiration
+
+Largely inspired by the settings model that both VS Code (and Sublime Text) use.
+
+### Goal: Minimize Writing to `profiles.json`
+
+We want to write the user settings file, `profiles.json`, as little as possible.
+Each time we write the file, there's the possiblity that we've re-ordered the
+keys, as `jsoncpp` provides no ordering guarantee of the keys. This isn't great,
+as each write of the file will randomly re-order the file.
+
+One of our overarching goals with this change should be to re-write the user
+settings file as little as possible.
+
+## Solution Design
+
+The settings are now composed from two files: a "Default" settings file, and a
+"User" settings file.
+
+### Default Settings
+
+We'll have a static version of the "Default" file hardcoded within the
+application. We'll write the hardcoded version of this file out to an actual
+file on disk, `defaults.json`, however, that file will not actually be used. The
+defaults file will contain some sort of disclaimer like:
+
+```json
+// This is an auto-generated file. Place any modifications to your settings in "profiles.json"
+```
+
+This disclaimer will help identify that the file shouldn't be modified.
+
+The `defaults.json` file will be written out whenever we detect that it doesn't exist.

```

Comment:
Must we? How does VSCode handle it? Is their `defaults.json` in the user's owned directories or is it read-only living in a package-owned directory?

Reply:
I don't know if it exists _anywhere_ actually. 
For the user settings, you can reveal them in explorer:
![image](https://user-images.githubusercontent.com/18356694/59535822-8b601500-8eb7-11e9-938a-ff5bc78541fb.png)


But you can't for the defaults:
![image](https://user-images.githubusercontent.com/18356694/59535838-974bd700-8eb7-11e9-8405-34d4d8604d1b.png)

Both VSCode and Sublime completely prevent you from editing the file, and the path sublime shows for it's defaults actually _doesn't exist_.

I also searched for the `defaultSettings.json` file in the VsCode install, and it definitely doesn't exist there

Reply:
@zadjii-msft I think you are correct about it [not](https://github.com/microsoft/vscode/issues/52050) existing.     Their settings code is pretty interesting.  Worth a look.